src/Resolve/graphtype.jl

```diff
@@ -570,7 +570,13 @@ function wipe_snapshots!(graph::Graph)
     return graph
 end
 
-color_string(c, str...) = string(Base.text_colors[c], str..., Base.text_colors[:default])
+function color_string(c, str...)
+    if stderr[:color]  #check if color has been disabled by `julia --color=no`
```

Comment:
The `color` attribute on the io is what determines color support AFAIU.

Also, might be better to use `sprint` with `printstyled` instead of relying on internals (like `text_color`).

Reply:
I initially tried to use `sprint` + `printstyled` but it doesn't work.
I guess related to the properties of the IO Object it uses.

![image](https://user-images.githubusercontent.com/5127634/84690510-81de1280-af3a-11ea-957b-5070a5a8d10a.png)

Maybe I can create a different one that does support it?

> The color attribute on the io is what determines color support AFAIU.

Yes. but I do not have the `io` object, because it wasn't passed into all the logging functions.
So I used `stderr` which supports the this attribute also and is where the logs eventually get written. (I checked it with julia run with and without `--color=no`, it works)

Reply:
>  I guess related to the properties of the IO Object it uses.

You need to pass `context` to `sprint` as well.